components/transport-jms/src/test/java/org/wso2/transport/jms/test/client/polling/QueueClientPollingWithSelectorTestCase.java

```diff
@@ -0,0 +1,185 @@
+import org.wso2.transport.jms.test.util.JMSTestConstants;
+import org.wso2.transport.jms.test.util.JMSTestUtils;
+import org.wso2.transport.jms.utils.JMSConstants;
+
+import java.util.Map;
+import javax.jms.Connection;
+import javax.jms.DeliveryMode;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+/**
+ * Test case for queue polling with selectors.
+ */
+public class QueueClientPollingWithSelectorTestCase {
+    private static final Logger logger = LoggerFactory.getLogger(QueueTopicAutoAckListeningTestCase.class);
+
+    private final String queueName = "pollingQueue";
+
+    private JMSServer jmsServer;
+
+    private Map<String, String> jmsClientParameters;
+
+    private JMSClientConnector jmsClientConnectorQueue;
+
+    private Connection connection;
+
+    private String correlationId1 = "12345abcd";
+
+    private String correlationId2 = "12345xyz";
+
+    /**
+     * Starts the JMS Server, and publish messages.
+     *
+     * @throws JMSConnectorException Client Connector Exception.
+     */
+    @BeforeClass(groups = "jmsPolling",
+                 description = "Setting up the server, publish messages")
+    public void setUp() throws JMSConnectorException, InterruptedException, JMSException {
+        jmsClientParameters = JMSTestUtils.
+                createJMSParameterMap(JMSTestConstants.QUEUE_NAME, JMSTestConstants.QUEUE_CONNECTION_FACTORY,
+                        JMSConstants.DESTINATION_TYPE_QUEUE, JMSConstants.AUTO_ACKNOWLEDGE_MODE);
+        jmsServer = new JMSServer();
+        jmsServer.startServer();
+
+        // this will make the broker start and keep on running until the test is over
+        connection = jmsServer.createConnection(null, null);
+        connection.start();
+
+        //publish 10 messages to the queue with correlationId1
+        publishToQueue(correlationId1, 10);
+        //publish 5 messages to the queue with correlationId2
+        publishToQueue(correlationId2, 5);
+
+        jmsClientConnectorQueue = new JMSConnectorFactoryImpl().createClientConnector(jmsClientParameters);
+    }
+
+    /**
+     * This test will publish 10 messages to the provided queue and consume them using JMS transport client polling
+     * interface and validate the number of produced messages.
+     *
+     * @throws JMSConnectorException  Error when polling messages through JMS transport connector.
+     * @throws JMSException           Error when running the test message consumer.
+     * @throws NoSuchFieldException   Error when accessing the private field.
+     * @throws IllegalAccessException Error when accessing the private field.
+     */
+    @Test(groups = "jmsPolling",
+          description = "Queue Polling with selectors test case")
+    public void queuePollingTestCase()
+            throws JMSConnectorException, NoSuchFieldException, IllegalAccessException, InterruptedException,
```

Comment:
Should it be enough to throw Exception instead throwing specific exceptions as this is a test case?

Reply:
won't it be helpful to debug failures of test cases